### PR TITLE
feature/existing user sign in

### PR DESCRIPTION
**Pull request by MTCawthray**
# Thank you for pushing your code!
## What (if any) features are you implementing?
- This branch sets up the user flow of existing users being able to sign into their accounts.
- Now, when a user returns to the app after already setting up their profile, they can simply sign in with their username and password to get into their account.
- If a user enters an incorrect username or password, they will see error handling to tell them to check their inputs.

## What (if anything) did you refactor?
- I had to create a brand new ```SignIn``` component in order to use the ```loginUser``` mutation.

## Were there any issues that arose?
- Async issues with the data coming in from the mutation. I had to get creative with the setTimeout() function.

## Is there anything that you need from your teammates?
- NA

## Any other comments, questions, or concerns?
- This completes the login user flows! Now we just need to style, test, test, test, test, and test!

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor

## How Has This Been Tested?
- [x] open index.html
- [x] console.log()
- [x] dev tools
- [ ] Cypress

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/SignIn/SignIn.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  if (data) {
+    setTimeout(() => {handleLoad(data)}, 1000)
+  }
+
+  const handleLoad = (data) => {
+    if (data.loginUser.success){
+      assignUser(data.loginUser.id);
+      history.push(`/profile/${data.loginUser.id}`); 
+      console.log("data in sign in", data);
+    } else {
```
Comment (nataliemcintyre2021) [inline src/components/SignIn/SignIn.js:24]: Error handling looking good!

File: src/components/SignIn/SignIn.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+          onChange={(e) => setUsername(e.target.value)}
+          placeholder="enter your username"
+        />
+
+        <input
+          className="login-input"
+          type="password"
+          value={password} onChange={(e) => setPassword(e.target.value)}
+          placeholder="enter your password"
+          minlength={8} required
```
Comment (nataliemcintyre2021) [inline src/components/SignIn/SignIn.js:58]: Nice use of min length and required for error handling.

File: src/components/Login/Login.js [hunk 3]
```diff
@@ -1,10 +1,9 @@
 import './Login.css';
 import React, { useState, useEffect } from 'react';
-import {CREATE_USER} from '../../graphQL/mutations';
+import {CREATE_USER, LOGIN_USER} from '../../graphQL/mutations';
```
Comment (nataliemcintyre2021) [inline src/components/Login/Login.js:3]: Would love to talk through the use of both mutations in this component - awesome!

**Review by nataliemcintyre2021 — APPROVED** [review]

File: src/components/SignIn/SignIn.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
+import React, {useState, useEffect} from 'react';
```
Comment (hayleyw7) [inline src/components/SignIn/SignIn.js:1]: Thanks for getting this all moved to a new component. Best fix for our problem.

**Review by hayleyw7 — APPROVED** [review]
Great job!!!!

**Pull request merged**